src/EnergyPlus/OutputProcessor.cc

```diff
@@ -2735,89 +2601,66 @@ namespace OutputProcessor {
         }
     } // WriteCumulativeReportMeterData()
 
-    void WriteReportMeterData(EnergyPlusData &state,
-                              int const reportID,      // The variable's report ID
-                              Real64 const repValue,   // The variable's value
-                              ReportFreq const freq,   // The variable's reporting interval (e.g., hourly)
-                              Real64 const minValue,   // The variable's minimum value during the reporting interval
-                              int const minValueDate,  // The date the minimum value occurred
-                              Real64 const MaxValue,   // The variable's maximum value during the reporting interval
-                              int const maxValueDate,  // The date of the maximum value
-                              bool const meterOnlyFlag // Indicates whether the data is for the meter file only
+    void MeterPeriod::WriteReportData(EnergyPlusData &state,
+                                      ReportFreq const freq
     )
     {
 
         // SUBROUTINE INFORMATION:
         //       AUTHOR         Greg Stark
         //       DATE WRITTEN   July 2008
-        //       MODIFIED       na
-        //       RE-ENGINEERED  na
 
         // PURPOSE OF THIS SUBROUTINE:
         // This subroutine writes for the non-cumulative meter data to the output files and
         // SQL database.
 
-        std::string NumberOut; // Character for producing "number out"
         auto &sql = state.dataSQLiteProcedures->sqlite;
 
-        if (repValue == 0.0) {
+        std::string NumberOut;
+
+        if (Value == 0.0) {
             NumberOut = "0.0";
         } else {
-            char meterData[129];
-            dtoa(repValue, meterData);
-            NumberOut = std::string(meterData);
+           char tmp[128];
```

Comment:
Ok, I ran into this a couple of times and this is annoying and interferes with refactoring. Maybe this is not the best place to explain the problem.